doc/languages-frameworks/typst.section.md

```diff
@@ -0,0 +1,55 @@
+# Typst {#typst}
+
+Typst can be configured to include packages from Typst Universe or customized packages.
+
+## Custom Environment {#typst-custom-environment}
+
+Creating a custom Typst environment with a selected set of packages from Typst Universe can be done with the following code. It is possible to specify a specific version of a Typst package (e.g., `cetz_0_3_0`). A package without a version number refers to its latest version.
+
+```nix
+typst.withPackages (p: with p; [
+  polylux
+  cetz_0_3_0
+])
+```
+
+Since Typst Universe doesn't provide a way to fetch a package with a specific hash, it can be common that Typst packages' hash in nixpkgs are outdated. The example below can fix this issue.
+
+```nix
+typst.withPackages.override (old: {
+  typstPackages = old.typstPackages // {
+    polylux = old.typstPackages.polylux.overrideAttrs (oldPolylux: {
+      src = fetchurl {
+        url = oldPolylux.src.url;
+        hash = yourNewPolyluxHash;
+      };
+    });
+  };
+}) (p: with p; [
+  polylux
+  cetz_0_3_0
+])
+```
+
+## Custom Packages {#typst-custom-packages}
+
+Nixpkgs provide a helper `buildTypstPackage` to build custom Typst packages understood by the Typst environment. Any dependency of the custom package is required to be manually specified in `buildTypstpackage`. Here is how to do it.
+
+```nix
+{ buildTypstPackage, typstPackages, fetchurl }:
+
+buildTypstPackage {
+  pname = "my-typst-package";
+  version = "0.0.1";
+  src = fetchurl { ... };
+  typstDeps = with typstPackages; [ cetz_0_3_0 ];
+}
+```
+
+Note that every custom package is scoped in `@preview`, as shown below. To not violating the semantics of `@preview`, it is recommended to use this only for patching existing packages from Typst Universe.
```

Comment:
```suggestion
Note that every custom package is scoped in `@preview`, as shown below. In order to avoid violating the semantics of `@preview`, it is recommended to use this only for patching existing packages from Typst Universe.
```

I am not sure if "To not violating" is grammatically correct.